src/devices/Sht3x/Sht3x.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.I2c;
+using System.Threading;
+
+namespace Iot.Device.Sht3x
+{
+    /// <summary>
+    /// Humidity and Temperature Sensor SHT3x
+    /// </summary>
+    public class Sht3x : IDisposable
+    {
+        private I2cDevice _sensor;
+
+        // CRC const
+        private const byte CRC_POLYNOMIAL = 0x31;
+        private const byte CRC_INIT = 0xFF;
+
+        #region prop
+
+        /// <summary>
+        /// SHT3x Resolution
+        /// </summary>
+        public Resolution Resolution { get; set; }
+
+        private double _temperature;
+        /// <summary>
+        /// SHT3x Temperature (℃)
+        /// </summary>
+        public double Temperature { get { ReadTempAndHumi(); return _temperature; } }
+
+        private double _humidity;
+        /// <summary>
+        /// SHT3x Relative Humidity (%)
+        /// </summary>
+        public double Humidity { get { ReadTempAndHumi(); return _humidity; } }
+
+        private bool _heater;
+        /// <summary>
+        /// SHT3x Heater
+        /// </summary>
+        public bool Heater
+        {
+            get => _heater;
+            set
+            {
+                SetHeater(value);
+                _heater = value;
+            }
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Creates a new instance of the SHT3x
+        /// </summary>
+        /// <param name="sensor">I2C Device, like UnixI2cDevice or Windows10I2cDevice</param>
+        /// <param name="resolution">SHT3x Read Resolution</param>
+        public Sht3x(I2cDevice sensor, Resolution resolution = Resolution.High)
+        {
+            _sensor = sensor;
+
+            Resolution = resolution;
+
+            Reset();
+        }
+
+        /// <summary>
+        /// Cleanup
+        /// </summary>
+        public void Dispose()
+        {
+            _sensor?.Dispose();
+            _sensor = null;
+        }
+
+        /// <summary>
+        /// SHT3x Soft Reset
+        /// </summary>
+        public void Reset()
+        {
+            Write(Register.SHT_RESET);
+        }
+
+        /// <summary>
+        /// Set SHT3x Heater
+        /// </summary>
+        /// <param name="isOn">Heater on when value is true</param>
+        private void SetHeater(bool isOn)
+        {
+            if (isOn)
+                Write(Register.SHT_HEATER_ENABLE);
+            else
+                Write(Register.SHT_HEATER_DISABLE);
+        }
+
+        /// <summary>
+        /// Read Temperature and Humidity
+        /// </summary>
+        private void ReadTempAndHumi()
```

Comment:
does it make sense to split this into two functions (for reading temperature and humidity separately)? you call for both of them either way so feels like there is no benefit of having them together

Reply:
If I could use two methods, I would use them. This sensor is strange. To read the data, I need to send an address containing four hexadecimal bits, and then the sensor will directly return six bytes of data. In other words, temperature and humidity return together.

Reply:
haha 😄 won't argue - this is hardware - there are weird things here everywhere!